File: client/src/entities/Message.ts
import PageResponse from "./PageResponse";
import { UserData } from "./User";

export default interface MessageModel {
  messageId: number;
  message: string;
  messageUpdate: string;
  timestamp: string;
  chatId: number;
  sender: UserData;
}

export default interface MessageResponse {
  messageModels: MessageModel[];
  pageResponse: PageResponse;
}

File: server/src/main/java/com/facebook/server/entity/Message.java
package com.facebook.server.entity;

import jakarta.persistence.*;
import lombok.*;

@Builder
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "message")
public class Message extends Timestamp {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long messageId;
    private String message;
    private String messageUpdate;

    @ManyToOne
    @JoinColumn(name = "chat_id")
    private Chat chat;

    @ManyToOne
    @JoinColumn(name = "sender_id", nullable = false)
    private User sender;

}
